### PR TITLE
Fix error reporting in daml build, and improve daml clean messages (3x)

### PR DESCRIPTION
**Pull request by samuel-williams-da**
Forward port #19803

### CODE REVIEW

File: sdk/compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 1]
```diff
@@ -1288,38 +1288,48 @@ execClean :: ProjectOpts -> EnableMultiPackage -> MultiPackageLocation -> MultiP
...
     when isProject $ do
-        let removeAndWarn path = do
-                whenM (doesDirectoryExist path) $ do
-                    putStrLn ("Removing directory " <> path)
-                    removePathForcibly path
-                whenM (doesFileExist path) $ do
-                    putStrLn ("Removing file " <> path)
-                    removePathForcibly path
-        removeAndWarn damlArtifactDir
-        putStrLn "Removed build artifacts."
```
Comment (dylant-da) [inline sdk/compiler/damlc/lib/DA/Cli/Damlc.hs:1321]: Probably good to keep the last line "Removed build artifacts" that confirms removal is complete,

**Review by dylant-da — APPROVED** [review]
LGTM

File: sdk/compiler/damlc/lib/DA/Cli/Damlc.hs (hunk 1 above)
Comment (samuel-williams-da) [reply sdk/compiler/damlc/lib/DA/Cli/Damlc.hs:1321]: I feel a log of
```
Running multi-package clean of all packages in <path>
Cleaning <path1>
Removed build artifacts.
Cleaning <path2>
Removed build artifacts.
Cleaning <path3>
Removed build artifacts.
```
is a little weird, I guess I could just have one at the end?

Comment (dylant-da) [reply sdk/compiler/damlc/lib/DA/Cli/Damlc.hs:1321]: Yeah just one at the end is ideal

**Pull request merged**